velox/common/memory/MmapAllocator.cpp

```diff
@@ -397,8 +397,7 @@ bool MmapAllocator::allocateContiguousImpl(
           0);
     }
   }
-  // TODO: add handling of MAP_FAILED.
-  if (data == nullptr) {
+  if (data == nullptr || data == MAP_FAILED) {
```

Comment:
it might be useful to also print  the errno to the errorMsg when `data == MAP_FAILED` using `folly::errnoStr(errno)`

Reply:
@xiaoxmeng I saw you have imported this pull request. Can I update this PR directly to include what @bikramSingh91 suggested?